samples/core/src/test/java/io/oasp/gastronomy/restaurant/salesmanagement/service/impl/rest/SalesmanagementHttpRestServiceTest.java

```diff
@@ -61,6 +63,9 @@
   private SalesmanagementRestServiceTestHelper helper;
 
   @Inject
+  private ServiceClientFactory serviceClientFactory;
```

Comment:
The design of `ServiceClientFactory` is intended the way it is in order to be generic and flexible. Using generics to get information from it is very tricky due to erasure so you can not actually retrieve the type except via wild reflection tricks where you would have to know the name of the field and still a lot of complex stuff would need to be done. I went through all this hell when implementing support for it in m-m-m:
https://m-m-m.github.io/maven/apidocs/net/sf/mmm/util/reflect/api/GenericType.html
So we could create a new interface for this semantic with a non-arg `create()` method and use it as you propose. However, this would require that you get an individual instance injected and that depends on reflective implementation of the underlying bean (that might already be weaved with AspectJ). To sum up: Nice idea but would require much magic, will be fragile and might break with the next spring release, and in the end is not even shorter (ok, you actually save 3 characters in total).